src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/RazorCSharpLanguageMiddleware.ts

```diff
@@ -53,20 +54,25 @@ export class RazorCSharpLanguageMiddleware implements LanguageMiddleware {
                     documentUri);
 
                 if (!remappedResponse || !remappedResponse.range) {
-                    // Something went wrong when re-mapping to the original document. Ignore this edit.
-                    this.logger.logWarning(`Unable to remap file ${uri.path} at ${edit.range}.`);
-                    continue;
+                    const [codeActionUri, codeActionEdit] = this.tryApplyingCodeActions(uri, edit);
```

Comment:
Looks like we're trying to handle each code action edit individually. This design seems too restrictive because I can imagine code actions that require understanding of multiple edits to make sense of it. We might need to merge them in to a single edit or do some other magic to transform the original edits.

Reply:
From the meeting yesterday our current thought is that we should ape the behavior from EditAndContinue (wherein they get context about where in a syntax tree and edit is being made) and use that, rather than context about the codeAction to decide how to handle things.

I have a followup about that today, will comment here if my understanding changes.